### PR TITLE
Fixed issue #16

### PR DESCRIPTION
**Pull request by nesterov-n**
This reffers #16

### CODE REVIEW

**Comment by dmarkov** [comment]
@nesterov-n Thanks, I will find someone to review this PR soon

**Comment by dmarkov** [comment]
@longtimeago please review this one

File: src/test/java/com/jcabi/immutable/ArrayTest.java [hunk 1]
```diff
@@ -197,4 +199,16 @@ public void makesSubList() throws Exception {
...
+    /**
+     * Changing array we passed to ctor doesn't affect immutable array content.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void implicitImmutable() throws Exception {
+        final Integer[] ints = new Integer[] {1, 2, 3};
+        final Array<Integer> array = new Array<Integer>(ints);
+        ints[1] = 0;
+        Assert.assertEquals(2, (int) array.get(1));
```
Comment (longtimeago) [inline src/test/java/com/jcabi/immutable/ArrayTest.java]: @nesterov-n here you could use `java.util.Arrays.equals()`

File: src/test/java/com/jcabi/immutable/ArrayTest.java [hunk 2]
```diff
@@ -197,4 +199,16 @@ public void makesSubList() throws Exception {
         );
     }
 
+    /**
+     * Changing array we passed to ctor doesn't affect immutable array content.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void implicitImmutable() throws Exception {
```
Comment (longtimeago) [inline src/test/java/com/jcabi/immutable/ArrayTest.java]: @nesterov-n test name should be changed. See http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names

**Comment by longtimeago** [comment]
@nesterov-n please, see 2 remarks above

**Comment by nesterov-n** [comment]
@longtimeago Thank for remarks. Please, see my commit

File: src/test/java/com/jcabi/immutable/ArrayTest.java [hunk 3]
```diff
@@ -197,4 +199,18 @@ public void makesSubList() throws Exception {
...
+    /**
+     * Array must be independent from data we passed to its ctor.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void isIndependentFromCtorParam() throws Exception {
+        final Integer[] ints = new Integer[] {1, 2, 3};
+        final Array<Integer> array = new Array<Integer>(ints);
+        ints[1] = 0;
+        final Object[] actual = array.toArray();
```
Comment (longtimeago) [inline src/test/java/com/jcabi/immutable/ArrayTest.java]: @nesterov-n do we need these variables? Let's inline them

**Comment by longtimeago** [comment]
@nesterov-n one more remark

**Comment by nesterov-n** [comment]
@longtimeago fixed

File: src/test/java/com/jcabi/immutable/ArrayTest.java [hunk 4]
```diff
@@ -197,4 +199,19 @@ public void makesSubList() throws Exception {
...
+     * @throws Exception If some problem inside
+     * @checkstyle MagicNumberCheck (10 lines)
+     */
+    @Test
+    public void isIndependentFromCtorParam() throws Exception {
+        final Integer[] ints = new Integer[] {1, 2, 3};
+        final Array<Integer> array = new Array<Integer>(ints);
+        ints[1] = 0;
+        Assert.assertTrue(
+                Arrays.equals(array.toArray(), new Integer[]{1, 2, 3})
```
Comment (longtimeago) [inline src/test/java/com/jcabi/immutable/ArrayTest.java]: @nesterov-n I might be wrong, but the indentation is wrong here. Please, run build with `-Pqulice` to check

**Comment by longtimeago** [comment]
@nesterov-n thanks!

**Comment by longtimeago** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmarkov** [comment]
@longtimeago done, I added **22 mins** in payment `80573952`, 95 hours and 6 mins spent to complete this. extra minutes for review comments (c=7). +22 added to your rating, current score is: [+7368](http://www.netbout.com/b/35024?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy pls

**Pull request merged**